src/modules/ims_isc/ims_isc_mod.h

```diff
@@ -77,6 +77,29 @@ enum dialog_direction
 	/** Unknown 	*/
 };
 
+/* Method of obtaining isc_mark
+First bit -> 0=REQUEST_ROUTE/1=FAILURE_ROUTE 
+Second bit -> 0=Found/1=Missing
+*/
+enum isc_mark_status
+{
+	ISCMARK_FOUND_ROUTE_HEADER =
+			0, /*Request has been received from AS, old ISCMARK found in Route header field*/
+	ISCMARK_FOUND_LUMPS =
+			1, /*Request to AS has been rejected or timed out, old ISCMARK found in lumps*/
+	ISCMARK_MISSING_START_TRIGGERING =
+			2, /*Request has been received without old ISCMARK in Route header field*/
+	ISCMARK_MISSING_START_TRIGGERING_SAR =
+			3 /*SAR/SAA has happened with HSS, due to terminating request (no old ISCMARK)*/
+};
+
+/* ISCMARK Status bits*/
+
+/* ISCMARK is obtained in failure route*/
+#define ISCMARK_FAILURE (1 << 0)
+/* ISCMARK could not be found */
+#define ISCMARK_MISSING (1 << 1)
```

Comment:
Pretty clear now. Anyway, maybe to further reduce the cognitive load (I'm getting old :stuck_out_tongue_closed_eyes:), I'd suggest writing it like this.

```suggestion
/* ISCMARK Status bits*/

/* ISCMARK is obtained in failure route*/
#define ISCMARK_FAILURE (1 << 0)
/* ISCMARK could not be found */
#define ISCMARK_MISSING (1 << 1)

enum isc_mark_status
{
	/** Request has been received from AS, old ISCMARK found in Route header field */
	ISCMARK_FOUND_ROUTE_HEADER = 0, 
	/** Request to AS has been rejected or timed out, old ISCMARK found in lumps */
	ISCMARK_FOUND_LUMPS = ISCMARK_FAILURE, 
	/** Request has been received without old ISCMARK in Route header field*/
	ISCMARK_MISSING_START_TRIGGERING =	ISCMARK_MISSING, 
	/** SAR/SAA has happened with HSS, due to terminating request (no old ISCMARK) */
	ISCMARK_MISSING_START_TRIGGERING_SAR = ISCMARK_MISSING|ISCMARK_FAILURE 
};

```